test/integration.js

```diff
@@ -3,102 +3,101 @@ import process from 'node:process';
 import {$} from 'execa';
-import {deleteAsync} from 'del';
-import * as gitUtil from '../source/git-util.js';
+import {temporaryDirectoryTask} from 'tempy';
+import {writePackage} from 'write-pkg';
 import * as util from '../source/util.js';
 
-test.before(async t => {
-	await fs.emptyDir('integration');
-	process.chdir('integration');
+const createEmptyGitRepo = async ($$, temporaryDir) => {
+	await $$`git init`;
 
-	await $`git init`;
-	await t.throwsAsync(gitUtil.latestTag(), undefined, 'prerequisites not met: repository should not contain any tags');
+	// `git tag` needs an initial commit
+	await fs.createFile(path.resolve(temporaryDir, 'temp'));
+	await $$`git add temp`;
+	await $$`git commit -m "init1"`;
+	await $$`git rm temp`;
+	await $$`git commit -m "init2"`;
+};
 
-	await fs.createFile('temp');
-	await $`git add .`;
-	await $`git commit -m 'init'`;
-	await deleteAsync('temp');
-});
+const createIntegrationTest = async (t, assertions) => {
+	await temporaryDirectoryTask(async temporaryDir => {
+		const $$ = $({cwd: temporaryDir});
 
-test.after.always(async () => {
-	process.chdir('..');
-	await deleteAsync('integration');
-});
+		await createEmptyGitRepo($$, temporaryDir);
+		process.chdir(temporaryDir);
 
-test.afterEach.always(async t => {
-	if (typeof t.context.teardown === 'function') {
-		await t.context.teardown();
-	}
-});
+		t.context.createFile = async file => fs.createFile(path.resolve(temporaryDir, file));
+		await assertions($$, temporaryDir);
+	});
+};
 
-test.serial('files to package with tags added', async t => {
-	await $`git tag v0.0.0`;
-	await fs.createFile('new');
-	await fs.createFile('index.js');
-	await $`git add new index.js`;
-	await $`git commit -m "added"`;
-
-	t.context.teardown = async () => {
-		await $`git rm new`;
-		await $`git rm index.js`;
-		await $`git tag -d v0.0.0`;
-		await $`git commit -m "deleted"`;
-	};
-
-	t.deepEqual(
-		await util.getNewFiles({files: ['*.js']}),
-		{unpublished: ['new'], firstTime: ['index.js']},
-	);
-});
+test('main', async t => {
+	await createIntegrationTest(t, async $$ => {
+		await t.context.createFile('testFile');
 
-test.serial.failing('file `new` to package without tags added', async t => {
-	await fs.createFile('new');
-	await fs.createFile('index.js');
+		const {stdout} = await $$`git status -u`;
 
-	t.context.teardown = async () => {
-		await deleteAsync(['new', 'index.js']);
-	};
+		t.true(
+			stdout.includes('Untracked files') && stdout.includes('testFile'),
+			'File wasn\'t created properly!',
+		);
+	});
+});
 
-	t.deepEqual(
-		await util.getNewFiles({files: ['index.js']}),
-		{unpublished: ['new'], firstTime: ['index.js']},
-	);
+const createFixture = test.macro(async (t, pkgFiles, commands, {unpublished, firstTime}) => {
+	await createIntegrationTest(t, async ($$, temporaryDir) => {
+		// /** @type {import('../source/util.js')} */
+		// const util = await esmock('../source/util.js', {}, {
+		// 	'node:process': {cwd: () => temporaryDir},
+		// });
```

Comment:
I'm trying to stub `process.cwd()` so the integration tests can be run concurrently (and without `--no-worker-threads`), but I'm getting a cryptic error message. This should be the only "blocker"